spring-test/src/main/java/org/springframework/test/web/servlet/htmlunit/HtmlUnitRequestBuilder.java

```diff
@@ -291,9 +292,7 @@ private void cookies(MockHttpServletRequest request) {
 
 		Cookie[] parentCookies = request.getCookies();
 		if (parentCookies != null) {
-			for (Cookie cookie : parentCookies) {
-				cookies.add(cookie);
-			}
+			cookies.addAll(Arrays.asList(parentCookies));
```

Comment:
Please avoid the creation of the extra `ArrayList` as follows.

```suggestion
			Collections.addAll(cookies, parentCookies);
```